integration/internal/process.go

```diff
@@ -105,14 +115,37 @@ func (ps *ProcessState) Start(stdout, stderr io.Writer) (err error) {
 	}
 
 	select {
-	case <-detectedStart:
+	case <-ready:
 		return nil
 	case <-timedOut:
-		ps.Session.Terminate()
+		if ps.Session != nil {
+			ps.Session.Terminate()
+		}
 		return fmt.Errorf("timeout waiting for process %s to start", path.Base(ps.Path))
 	}
 }
 
+func safeMultiWriter(writers ...io.Writer) io.Writer {
+	safeWriters := []io.Writer{}
+	for _, w := range writers {
+		if w != nil {
+			safeWriters = append(safeWriters, w)
+		}
+	}
+	return io.MultiWriter(safeWriters...)
+}
+
+func pollURLUntilOK(url url.URL, ready chan bool) {
+	for {
+		res, err := http.Get(url.String())
+		if err == nil && res.StatusCode == http.StatusOK {
+			ready <- true
+			return
+		}
+		time.Sleep(time.Millisecond * 100)
```

Comment:
Consider exposing the loop iteration interval as a configurable option alongside timeout to reduce the potential for negative interaction between interval and timeout (e.g. timeout < interval).

It might be worth cribbing from the apimachinery polling code:

https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L240

Reply:
👍